### ISSUE TITLE
If modules overlap, deleting one can clobber the other

### ISSUE THREAD

**Issue by FutureAIGuru** [FutureAIGuru/BrainSimII#154]
**Describe the bug**
If modules overlap, deleting one can clobber the other

**Application**: Brain Simulator,

**To Reproduce**
1. Create network with ImageFile Module
2. Overlay ColorComponent module over a pixel of the picture area
3. Delete the ImageFile module
4. The ColorComponent module crashes because it relies on neuron settings which have been removed

**Comment by Moorelife** [comment]
Maybe we are having trouble here since the module literally occupy neurons rather than having their own "sub-network". If every module had its own network they would be truly independent.

**Comment by FutureAIGuru** [comment]
Yes, this is also exactly the problem of dragging modules or selections across each other. It seems like this would be a big deal to fix but perhaps not.
